### PR TITLE
unify learning rates in EngineBase

### PR DESCRIPTION
**Pull request by JackTemaki**
learning rate handling is the same for TF and PT.

### CODE REVIEW

File: returnn/tf/engine.py [hunk 1]
```diff
@@ -877,20 +877,17 @@ def __init__(self, config=None):
         self.tf_session = None  # type: typing.Optional[tf.compat.v1.Session]
         self.network = None  # type: typing.Optional[TFNetwork]
         self.updater = None  # type: typing.Optional[Updater]
-        self.learning_rate_control = None  # type: typing.Optional[LearningRateControl]
         self._checked_uninitialized_vars = False
         self._merge_all_summaries = None
         self.dataset_batches = {}  # type: typing.Dict[str,BatchSetGenerator]
         self.dataset_provider = None  # type: typing.Optional[DatasetDataProvider]
         self.train_data = None  # type: typing.Optional[Dataset]
-        self.eval_datasets = {}  # type: typing.Dict[str,Dataset]
```
Comment (albertz) [inline returnn/tf/engine.py:886]: Why did you remove this?

File: returnn/tf/engine.py [hunk 2]
```diff
@@ -1080,6 +1078,7 @@ def init_train_from_config(self, config=None, train_data=None, dev_data=None, ev
         :param returnn.datasets.basic.Dataset|None dev_data:
         :param returnn.datasets.basic.Dataset|None eval_data:
         """
+        super().init_train_from_config(config=config)
```
Comment (albertz) [inline returnn/tf/engine.py:1081]: It should come after `if not config: config = self.config`.

File: returnn/tf/engine.py [hunk 3]
```diff
@@ -1091,6 +1089,7 @@ def init_train_from_config(self, config=None, train_data=None, dev_data=None, ev
             from returnn.datasets.basic import set_config_extern_data_from_dataset
 
             set_config_extern_data_from_dataset(config=config, dataset=train_data or dev_data or eval_data)
+        super().init_train_from_config(config=config)
```
Comment (albertz) [inline returnn/tf/engine.py:1092]: This should be as early as possible, i.e. right after `if not config: config = self.config`.

File: returnn/tf/engine.py (hunk 1 above)
Comment (JackTemaki) [reply returnn/tf/engine.py]: I went a little ahead and also checked for unification of datasets, and this was a leftover I missed when doing the PR.

**Comment by JackTemaki** [comment]
@albertz please do not send each comment one by one, but group them all and send them once, this makes it easier to fix it all at once and saves time.

**Comment by albertz** [comment]
Btw, this would have been better squashed instead of rebased and merged, as those commits all belong together. But this needs to be decided individually per PR.
